frontends/web/src/routes/account/walletconnect/components/incoming-pairing/incoming-pairing.tsx

```diff
@@ -61,12 +61,17 @@ export const WCIncomingPairing = ({
     setPairingLoading(true);
     try {
       const { id, params } = currentProposal;
-      const { requiredNamespaces } = params;
-      const eipList = Object.values(requiredNamespaces);
+      const { requiredNamespaces, optionalNamespaces } = params;
+      const eipList = Object.keys(requiredNamespaces).length !== 0 ? Object.values(requiredNamespaces) : Object.values(optionalNamespaces);
+      if (!eipList) {
+        alertUser('Connecting app request did not define necessary namespace');
+        await handleRejectPairing();
+        setPairingLoading(false);
+      }
       const accounts = eipList.flatMap(eip => eip.chains?.map(chain => `${chain}:${receiveAddress}`) || []);
-      // For supported chains, use an intersection of supported chains and required chains
+      // For supported chains, use an intersection of supported chains and required chains, default to mainnet if no chains present
       const chains: string[] = eipList.flatMap(proposal =>
-        proposal.chains ? proposal.chains.filter(chain => Object.keys(SUPPORTED_CHAINS).includes(chain)) : []
+        proposal.chains ? proposal.chains.filter(chain => Object.keys(SUPPORTED_CHAINS).includes(chain)) : ['eip155:1']
```

Comment:
👍 looks good, maybe one day we need to define the chain id as a constant (e.g `eip155:1` to be `ETHEREUM_MAINNET`) just for clarity. But fine for now as it's only used once this way here.